vpr/src/timing/timing_fail_error.cpp

```diff
@@ -0,0 +1,48 @@
+#include <fstream>
+#include <sstream>
+
+#include "vtr_log.h"
+#include "vtr_assert.h"
+#include "vtr_math.h"
+
+#include "globals.h"
+#include "timing_util.h"
+#include "timing_fail_error.h"
+
+/* Represents whether or not VPR should fail if timing constraints aren't met. */
+static bool terminate_if_timing_fails = false;
```

Comment:
I've changed the exit message to what you recommended and not have the exit message contain the slacks. I figure you're right-- the user can just scroll up a bit and see exactly what they are, so there's no use in printing them twice.

Reply:
Re storing in a context: in the old branch that this code was in (in June I messed up and deleted the original branch this feature was on by accident and had to make a new branch & PR), you left a comment that recommended I don't put the bool representing whether or not VPR should fail if timing isn't met in a context: 
![image](https://user-images.githubusercontent.com/66747201/182442209-c2d5cc6a-7e91-41ca-b9d0-8ea56d65dee1.png)
I can't view the code that this comment references (because I deleted the branch :( ), but line 100 in vpr_context.h is in the middle of the TimingContext definition:
![image](https://user-images.githubusercontent.com/66747201/182442535-6327c076-a680-4b9f-8d95-9380439e4e33.png)

Reply:
Thanks. The control variable (fail_status_if_fail_timing) is fine to have in a context (that makes sense in vpr context).  I don't like the failure string being built up over many calls in either a context variable or a file scope variable -- instead it would be better if it was just a local variable where you created the proper message and printed it right away, and then decided to stop processing.  Is that doable?

Reply:
If I didn't misinterpret your comment, the most recent commit should address your concerns. I changed the flow s.t. I'm no longer building up the failure string, but immediately throwing the error when the negative slack is encountered. This is more useful (I think) since the error now gives a helpful line number to trace the error back to so you know _exactly_ where it failed.